packages/react-native-reanimated/src/animation/util.ts

```diff
@@ -541,5 +541,9 @@ export function defineAnimation<
 export function cancelAnimation<T>(sharedValue: SharedValue<T>): void {
   'worklet';
   // setting the current value cancels the animation if one is currently running
+  if (sharedValue._initial != null) {
+    sharedValue.value = sharedValue._initial;
+    return;
+  }
```

Comment:
This is unnacceptable, animations on cancel would restore to their initial state:



https://github.com/user-attachments/assets/0b43e6b1-7378-4e96-91b1-2c333059aa15